tests/data/example_ode/kinetic_model.toml

```diff
@@ -1,83 +1,96 @@
-[[compartment]]
-id = 'c'
-name = 'cytosol'
-volume = 1
-
-[[metabolite-in-compartment]]
-id = "A_c"
-metabolite = 'A'
-name = 'Metabolite A'
-balanced = false
-compartment = 'c'
-
-[[metabolite-in-compartment]]
-id = "B_c"
-metabolite = 'B'
-name = 'Metabolite B'
-balanced = true
-compartment = 'c'
-
-[[metabolite-in-compartment]]
-id = "C_c"
-metabolite = 'C'
-name = 'Metabolite C'
-balanced = true
-compartment = 'c'
-
-[[metabolite-in-compartment]]
-id = "D_c"
-metabolite = 'D'
-name = 'Metabolite D'
-balanced = false
-compartment = 'c'
+compartment = [
+  {id = "c", name = "cytosol", volume = 1}
+]
 
+metabolite_in_compartment = [
+  {metabolite_id = "A", compartment_id = "c", balanced = false},
+  {metabolite_id = "B", compartment_id = "c", balanced = true},
+  {metabolite_id = "C", compartment_id = "c", balanced = true},
+  {metabolite_id = "D", compartment_id = "c", balanced = false},
+]
 
-[[reaction]]
-id = 'r1'
-name = 'reaction number 1'
-stoichiometry = { A_c = -1, B_c = 1}
-mechanism = "reversible_modular_rate_law"
-[[reaction.enzyme]]
-id = 'r1'
+enzyme_reaction = [
+  {enzyme_id = "r1", reaction_id = "r1"},
+  {enzyme_id = "r2A", reaction_id = "r2"},
+  {enzyme_id = "r2B", reaction_id = "r2"},
+  {enzyme_id = "r3", reaction_id = "r3"},
+  {enzyme_id = "r4", reaction_id = "r4"},
+]
+
+[[metabolite]]
+id = "A"
+name = "Metabolite A"
+
+[[metabolite]]
+id = "B"
+name = "Metabolite B"
+
+[[metabolite]]
+id = "C"
+name = "Metabolite C"
+
+[[metabolite]]
+id = "D"
+name = "Metabolite D"
+
+[[enzyme]]
+id = "r1"
 name = 'the enzyme that catalyses reaction r1'
-[[reaction.enzyme.modifier]]
-modifier_type = 'competitive_inhibitor'
-mic_id = 'D_c'
 
-[[reaction]]
-id = 'r2'
-name = 'reaction number 2'
-stoichiometry = { A_c = -1, C_c = 1 }
-mechanism = "reversible_modular_rate_law"
-[[reaction.enzyme]]
+[[enzyme]]
 id = 'r2A'
 name = 'Isoenzyme A that catalyses reaction r2'
-[[reaction.enzyme.modifier]]
-modifier_type = 'allosteric_activator'
-mic_id = 'C_c'
-[[reaction.enzyme]]
+
+[[enzyme]]
 id = 'r2B'
 name = 'Isoenzyme B that catalyses reaction r2'
-[[reaction.enzyme.modifier]]
-modifier_type = 'allosteric_inhibitor'
-mic_id = 'C_c'
 
+[[enzyme]]
+id = 'r3'
+name = 'the enzyme that catalyses reaction r3'
+
+[[enzyme]]
+id = 'r4'
+name = 'the enzyme that catalyses reaction r4'
+
+[[reaction]]
+id = "r1"
+name = "reaction number 1"
+stoichiometry = {A_c = -1, B_c = 1}
+mechanism = "reversible_michaelis_menten"
 
+[[reaction]]
+id = "r2"
+name = "reaction number 2"
+stoichiometry = {A_c = -1, C_c = 1}
+mechanism = "reversible_michaelis_menten"
```

Comment:
Is there any test case for the irreversible mechanism?

Reply:
No - I just made an issue #389